dbt/include/athena/macros/materializations/models/table/create_table_as.sql

```diff
@@ -32,14 +33,16 @@
       {%- endset -%}
       {% do exceptions.raise_compiler_error(error_unique_location_iceberg) %}
     {%- endif -%}
+  {%- elif external_location or s3_data_naming in ['table', 'schema_table'] -%}
```

Comment:
this might not cover all cases. e.g. if I set my s3_data_naming=table_unique and uses format=parquet.

Instead of adding another fork in the if/else why don't you add totally new section?

e.g. 

```
{%- if format != 'iceberg' -%}
  {% do adapter.prune_s3_table_location(location) %}
{%- endif %}
```
We want to delete the location in all the cases except for iceberg. I think that doing so the flow is a bit more readable.

Reply:
My thinking was that using `s3_data_naming` options that write to unique locations won't need to prune, as they are always new. Happy to change it to run on all non-iceberg formats though!